static/js/timerender.js

```diff
@@ -45,6 +45,39 @@ exports.render_now = function (time) {
     return [time.toString("MMM\xa0dd"), formal_time_str, false];
 };
 
+// Current date is passed as an argument for unit testing
+exports.last_seen_status_from_date = function (last_active_date, current_date) {
+    if (typeof  current_date === 'undefined') {
+         current_date = new XDate();
+    }
+
+    var minutes = Math.floor(last_active_date.diffMinutes(current_date));
+    if (minutes <= 2) {
+        return i18n.t("Last seen just now");
+    }
+    if (minutes < 60) {
+        return i18n.t("Last seen " + minutes +" minutes ago");
+    }
+
+    var hours = Math.floor(minutes / 60);
+    if (hours === 1) {
+         return i18n.t("Last seen an hour ago");
+    }
+    if (hours < 24) {
+        return i18n.t("Last seen " + hours + " hours ago");
+    }
+
+    var days = Math.floor(hours / 24);
+    if (days === 1) {
+        return [i18n.t("Last seen yesterday")];
+    }
+    if (days < 365) {
+        return i18n.t("Last seen on " + last_active_date.toString("MMM\xa0dd"));
```

Comment:
If current_date is 2017-01-01 and last_active_date is 2016-12-01, days will be less than 365 and the date string returned will not indicate the previous year (2016). I have made similar changes to timerender.render_now() in PR #4833.